src/main/java/uk/gov/justice/probation/courtcasematcher/service/ReplayHearingsService.java

```diff
@@ -113,17 +113,21 @@ private void processNewOrUpdatedHearing(String s3Path, String hearingId) {
         } catch (JsonProcessingException e) {
             throw new RuntimeException(e);
         }
-        final var hearing = cpHearingEvent.asDomain()
-            .withHearingId(cpHearingEvent.getHearing().getId())
-            .withHearingEventType("ConfirmedOrUpdated");
+        final var hearings = cpHearingEvent.asDomain()
+            .stream()
+            .map(h -> h.withHearingId(cpHearingEvent.getHearing().getId())
+                .withHearingEventType("ConfirmedOrUpdated"))
```

Comment:
I know you haven't added this line but it looks suspicious - will have a quick look at it